### ISSUE TITLE
What distance threshold to choose with anisotropic voxel spacing?

### ISSUE THREAD

**Issue by IlvaHou55** [pymedphys/pymedphys#1605]
Hi everyone,

I was wondering what distance threshold you are using when the voxel spacing is very anisotropic (e.g. in plane (xy) voxel spacing is 0.5mm and between slices (z) is 4 mm). 
What I did now is gave a vector as input of the distance threshold to deal with different thresholds in different planes, but it just outputted two arrays for each different threshold. 
I think it is not fair to use either of those, because the distance threshold in the z-direction in the one case is very loose and will give non-representative results.

The only solution I can now come up with is interpolate the volume so that the z-spacing matches the xy-spacing. However, this will drastically increase the number of points to be checked. It already took one entire day to calculate the gamma index now, imagine when the number of points is eight times as large.

Any ideas/tips?

Thank you!

**Comment by SimonBiggs** [comment]
Hi @IlvaHou55,

Thanks for reaching out :) as a first point around the speed taken, have you tried using the `max_gamma` parameter? Also, how different to each other are your two arrays? The more different, the longer it will take without using the `max_gamma` parameter.

To see all the parameters you can tweak have a look at the reference docs:

https://docs.pymedphys.com/lib/ref/gamma.html

You can also reduce the `interp_fraction` if that is desired.

The gamma distance metric is usually defined equally in all directions. If you would like to have an effectively different distance threshold in one direction I could imagine you could write a wrapper function which scaled the coordinate axis so as to result in some directions having a greater distance threshold than others. However, I have never seen gamma utilised in this way.

If speed is what you're after, and you only care about a passing metric, you can also do a "monte carlo" style gamma calc by using the `random_subset` parameter. You can set this to a certain number of points, say 1000. This will then randomly select only 1000 points and only run the gamma within that random subset. If you then repeat the calculation a few times and verify that it is consistent to what you want, you can get quite a quick passing rate value.

Cheers,
Simon

**Comment by IlvaHou55** [comment]
@SimonBiggs Thank you for your elaborate answer! Speed is not really the issue, but it would be if my volume increases due to interpolation between slices. I'll check your options and see what suits best for my application.

**Comment by Matthew-Jennings** [comment]
@IlvaHou55, perhaps you could try the new implementation in #1761 either now by cloning pymedphys and checking out the relevant branch or later once it lands?

**Comment by SimonBiggs** [comment]
@Matthew-Jennings, for user testing, you're actually free to create a dev release on PyPI at any point. (Increment the version number and add -dev0 on the end, run propagate, them poetry build and poetry publish).

I find it really helps in getting real world feedback of users can just install `pip install pymedphys[user]==0.x.x-devx`

**Comment by SimonBiggs** [comment]
Nice thing about a dev release on PyPI is that it is never installed by default. The user has to explicitly request the version (or use other PyPI options that explicitly allow dev installs)

**Comment by Matthew-Jennings** [comment]
Thanks @SimonBiggs - should I include an explicit CHANGELOG entry for 0.40.0-dev0 as well?

**Comment by SimonBiggs** [comment]
Nope, for all intensive purposes "it doesn't exist". Think of it as a "development branch" on PyPI.

**Comment by SimonBiggs** [comment]
An example recently where I found it was really helpful. A user had a bug during a video call. I was able to make a branch on PyMedPhys, and make a dev release straight from that branch (not even main), they were then able to download the fix, directly, by installing the dev version. We tested it out in the video call, and it all worked :).
